source/robot_model/src/Model.cpp

```diff
@@ -334,86 +307,80 @@ Model::inverse_geometry(const state_representation::CartesianState& desired_cart
   Eigen::MatrixXd W_c = Eigen::MatrixXd::Identity(this->robot_model_.nq, this->robot_model_.nq);
   Eigen::VectorXd psi(this->robot_model_.nq);
   Eigen::VectorXd err(6);
-  const state_representation::CartesianPose X_d(desired_cartesian_state);
-
-  int i = 0;
-  while (i < params.max_number_of_iterations) {
-    err = ((X_d - this->forward_geometry(q, frame_id)) / dt).data();
-    if (err.cwiseAbs().maxCoeff() < params.tolerance) {
+  for (unsigned int i = 0; i < parameters.max_number_of_iterations; ++i) {
+    err = ((cartesian_pose - this->forward_kinematics(q, frame_id)) / dt).data();
+    // in case of convergence directly get out
```

Comment:
instead of `directly get out` i would suggest
```suggestion
    // break in case of convergence
```